includes/wc-user-functions.php

```diff
@@ -204,46 +204,14 @@ function wc_customer_bought_product( $customer_email, $user_id, $product_id ) {
 		return $result;
 	}
 
-	$transient_name = 'wc_cbp_' . md5( $customer_email . $user_id . WC_Cache_Helper::get_transient_version( 'orders' ) );
-
-	if ( false === ( $result = get_transient( $transient_name ) ) ) {
-		$customer_data = array( $user_id );
-
-		if ( $user_id ) {
-			$user = get_user_by( 'id', $user_id );
-
-			if ( isset( $user->user_email ) ) {
-				$customer_data[] = $user->user_email;
-			}
-		}
-
-		if ( is_email( $customer_email ) ) {
-			$customer_data[] = $customer_email;
-		}
-
-		$customer_data = array_map( 'esc_sql', array_filter( array_unique( $customer_data ) ) );
-		$statuses      = array_map( 'esc_sql', wc_get_is_paid_statuses() );
-
-		if ( sizeof( $customer_data ) == 0 ) {
-			return false;
-		}
-
-		$result = $wpdb->get_col( "
-			SELECT im.meta_value FROM {$wpdb->posts} AS p
-			INNER JOIN {$wpdb->postmeta} AS pm ON p.ID = pm.post_id
-			INNER JOIN {$wpdb->prefix}woocommerce_order_items AS i ON p.ID = i.order_id
-			INNER JOIN {$wpdb->prefix}woocommerce_order_itemmeta AS im ON i.order_item_id = im.order_item_id
-			WHERE p.post_status IN ( 'wc-" . implode( "','wc-", $statuses ) . "' )
-			AND pm.meta_key IN ( '_billing_email', '_customer_user' )
-			AND im.meta_key IN ( '_product_id', '_variation_id' )
-			AND im.meta_value != 0
-			AND pm.meta_value IN ( '" . implode( "','", $customer_data ) . "' )
-		" );
-		$result = array_map( 'absint', $result );
-
-		set_transient( $transient_name, $result, DAY_IN_SECONDS * 30 );
-	}
-	return in_array( absint( $product_id ), $result );
+	$product_purchases = $wpdb->get_row( $wpdb->prepare( "
+		SELECT *
+		FROM {$wpdb->prefix}woocommerce_product_purchases
+		WHERE ( user_id = %s OR user_email = %s ) AND product_id = %d
+		LIMIT 1
+	", absint( $user_id ), esc_sql( $customer_email ), absint( $product_id ) ) );
+
+	return ! empty( $product_purchases );
```

Comment:
I'm wondering if it's wise to deprecate `woocommerce_customer_bought_product` in favor of `WC_Purchased_Products_Helper::is_product_purchased`

Reply:
I'm in favour of this, I feel that wc_customer_bought_product have too many params, it makes no sense specifying a user id and an email to check for a product purchase.

I'd make the new function only accept 2 params, user_id/email and product_id, you can do is_email and is_int on the first param to decide what to check on.